### DISCUSSION TITLE
How to update Path of boundingRect in Bezier curve

### DISCUSSION THREAD

**Discussion by seoneunhye** [fabricjs/fabric.js#9800] [Q&A] [answered]
```
    const onObjectMoving = (e) => {
      const { target } = e;

      const { curve } = target;
      if (!curve && target.name !== 'lineCurve') return;

      if (target.name === 'startPoint') {
        curve.path[0][1] = target.left;
        curve.path[0][2] = target.top;
        curve.set({ path: curve.path, dirty: true });
        curve.setCoords();
      } else if (target.name === 'curvePoint') {
        curve.path[1][1] = target.left;
        curve.path[1][2] = target.top;

        curve.set({ path: curve.path, dirty: true });
        curve.setCoords();
      } else if (target.name === 'endPoint') {
        curve.path[1][3] = target.left;
        curve.path[1][4] = target.top;
        curve.set({ path: curve.path, dirty: true });
        curve.setCoords();
      }

      fabricDrawingcanvas.value.renderAll();
    };
```

I'm working with startPoint, endPoint, and curvePoint, which means when I move these control points, the curve, or Bezier curve, moves accordingly. I'm using Vue 3 for this. I referred to this code: http://fabricjs.com/quadratic-curve

But when I move the control points and the length of the Path increases, the boundingRect doesn't update. What could be the problem?

**Comment by ShaMan123** [comment, marked as answer]
Path's API is not great
You need to call:

https://github.com/fabricjs/fabric.js/blob/6b6a5fc6f2a79bae70542611f18220d51e1480d7/src/shapes/Path.ts#L93-L96

And you will probably want to set the position to start/end/center of your bezier

**Reply by ShaMan123** [reply]
I would avoid using `adjustPosition`, it has a bug IIRC
